src/Microsoft.ML.Transforms/MissingValueReplacingUtils.cs

```diff
@@ -340,6 +362,52 @@ protected long GetValuesProcessed(int slot)
             protected abstract void ProcessValueMax(in TItem val, int slot);
         }
 
+        /// <summary>
+        /// A mutable struct for keeping the appropriate statistics for mode calculations, whose scope is restricted
+        /// and only exists as an instance in a field or an array, utilizing the mutation of the struct correctly.
+        /// </summary>
+        private class ModeStat<TType>
+        {
+            // Delegate used to check if the value is valid. We use a delegate so that this class can support modes of all types.
+            public delegate bool IsValid(in TType val);
+
+            private TType _modeSoFar;
+            private int _maxCount;
+            private Dictionary<TType, int> _valueCounts;
+            private IsValid _validityCheck;
+            public ModeStat(IsValid valid)
+            {
+                _modeSoFar = default;
+                _maxCount = 0;
+                _valueCounts = new Dictionary<TType, int>();
+                _validityCheck = valid;
+            }
+
+            public void Update(TType val)
+            {
+                // We don't include non finite values in the mode, so if its not finite then just return.
+                if (!_validityCheck(val))
+                    return;
+
+                int count = 1;
+                if (_valueCounts.TryGetValue(val, out int curCount))
+                    count = curCount + 1;
+
```

Comment:
nit: 
If TryGetValue fails it initializes the out variable to default = 0.  Maybe this is sufficient.
```
_valueCounts.TryGetValue(val, out int count);
count++;
```

Reply:
Good point. Code is updated.